app/controllers/application_controller.rb

```diff
@@ -58,4 +58,8 @@ def page_tracker
       PageTracker.new(trainee_slug: trainee_slug, session: session, request: request)
     end
   end
+
+  def training_route_manager
+    @training_route_manager ||= TrainingRouteManager.new(trainee)
```

Comment:
We probably shouldn't expose this here as we've no guarantee `trainee` exists.

Reply:
My latest commit has a different approach which @defong suggested.